gradoop-examples/src/main/java/org/gradoop/benchmark/sampling/SamplingBenchmark.java

```diff
@@ -0,0 +1,421 @@
+        return new RandomNonUniformVertexSampling(Float.parseFloat(CONSTRUCTOR_PARAMS[0]),
+          Long.parseLong(CONSTRUCTOR_PARAMS[1]));
+      } else {
+        throw createInstantiationException(Algorithm.RANDOM_NON_UNIFORM_VERTEX_SAMPLING.name,
+          new String[]{"1", "2"});
+      }
+
+    case RANDOM_VERTEX_EDGE_SAMPLING:
+      if (CONSTRUCTOR_PARAMS.length == 1) {
+        return new RandomVertexEdgeSampling(Float.parseFloat(CONSTRUCTOR_PARAMS[0]));
+      } else if (CONSTRUCTOR_PARAMS.length == 2) {
+        return new RandomVertexEdgeSampling(Float.parseFloat(CONSTRUCTOR_PARAMS[0]),
+          Float.parseFloat(CONSTRUCTOR_PARAMS[1]));
+      } else {
+        throw createInstantiationException(Algorithm.RANDOM_VERTEX_EDGE_SAMPLING.name,
+          new String[]{"1", "2"});
+      }
+
+    case RANDOM_VERTEX_NEIGHBORHOOD_SAMPLING:
+      if (CONSTRUCTOR_PARAMS.length == 1) {
+        return new RandomVertexNeighborhoodSampling(Float.parseFloat(CONSTRUCTOR_PARAMS[0]));
+      } else if (CONSTRUCTOR_PARAMS.length == 2) {
+        return new RandomVertexNeighborhoodSampling(Float.parseFloat(CONSTRUCTOR_PARAMS[0]),
+          Long.parseLong(CONSTRUCTOR_PARAMS[1]));
+      } else {
+        throw createInstantiationException(Algorithm.RANDOM_VERTEX_NEIGHBORHOOD_SAMPLING.name,
+          new String[]{"1", "2"});
+      }
+
+    case RANDOM_VERTEX_SAMPLING:
+      if (CONSTRUCTOR_PARAMS.length == 1) {
+        return new RandomVertexSampling(Float.parseFloat(CONSTRUCTOR_PARAMS[0]));
+      } else if (CONSTRUCTOR_PARAMS.length == 2) {
+        return new RandomVertexSampling(Float.parseFloat(CONSTRUCTOR_PARAMS[0]),
+          Long.parseLong(CONSTRUCTOR_PARAMS[1]));
+      } else {
+        throw createInstantiationException(Algorithm.RANDOM_VERTEX_SAMPLING.name,
+          new String[]{"1", "2"});
+      }
+
+    default:
+      throw  new IllegalArgumentException(
+        "Something went wrong. Please select an other sampling algorithm.");
+    }
+  }
+
+  /**
+   * Helper function to create string that can be used to display available associations between the
+   * ordinals of the enum Algorithm and the sampling algorithms themselves.
+   *
+   * @return String providing association information.
+   */
+  private static String getOrdinalAlgorithmAssociation() {
+    StringBuilder result = new StringBuilder("Available Algorithms:\n");
+    for (Algorithm alg : Algorithm.values()) {
+      result.append(String.format("%d ---> %s%n", alg.ordinal(), alg.name));
+    }
+    return result.toString();
+  }
+
+  /**
+   * Helper function used to create an exception that is specific to function
+   * instantiateSelectedSamplingAlgorithm()
+   *
+   * @param name Name of the sampling class.
+   * @param amountArgs Array of strings containing all possible argument counts for a given sampling
+   *                  class.
+   * @return IllegalArgumentException
+   */
+  private static IllegalArgumentException createInstantiationException(
```

Comment:
This Exception can be part of the builder as well.

Reply:
added exception to builder class